pkg/controller/endpointslice/reconciler.go

```diff
@@ -349,6 +352,10 @@ func (r *reconciler) reconcileByPortMapping(
 		// Fill the slice up with remaining endpoints.
 		for desiredSet.Len() > 0 && len(sliceToFill.Endpoints) < int(r.maxEndpointsPerSlice) {
 			endpoint, _ := desiredSet.PopAny()
+			if sliceToFill.Name != "" {
```

Comment:
should probably move this above this loop? I don't think we need to copy it every time

Reply:
yep, I'd fixed that locally but apparently not saved before commit :man_facepalming: